### PR TITLE
Remove wscale option from MLPConvolution2D

### PR DESCRIPTION
**Pull request by delta2323**
I found `wscale` option in `L.MLPConvolution2D` is not used any longer. Related to #2163

### CODE REVIEW

File: chainer/links/connection/mlp_convolution_2d.py [hunk 1]
```diff
@@ -44,8 +44,7 @@ class MLPConvolution2D(link.ChainList):
     """
 
     def __init__(self, in_channels, out_channels, ksize, stride=1, pad=0,
-                 activation=relu.relu, wscale=1,
-                 conv_init=None, bias_init=None):
+                 activation=relu.relu, conv_init=None, bias_init=None):
```
Comment (unnonouno) [inline chainer/links/connection/mlp_convolution_2d.py]: Add message that `wscale` argument is removed from v2.

Comment (delta2323) [inline chainer/links/connection/mlp_convolution_2d.py]: I am thinking how we should do. The problem in my mind is the following case:
`MLPConvolution2D(10, 3, 3, 1, 0, relu.relu, 1)`
It works without errors, although its semantic change. The problem is we cannot tell which API users expect.

Comment (unnonouno) [inline chainer/links/connection/mlp_convolution_2d.py]: How about using `kwargs` for all arguments to prohibit positional arguments?

Comment (delta2323) [inline chainer/links/connection/mlp_convolution_2d.py]: We need to make `conv_init` and `bias_init` keyword arguments and prohibit to make them positional ones. I think it is an acceptable compromisation.

File: chainer/links/connection/mlp_convolution_2d.py [hunk 2]
```diff
@@ -33,26 +36,44 @@ class MLPConvolution2D(link.ChainList):
...
-    """
+    """  # NOQA
 
     def __init__(self, in_channels, out_channels, ksize=None, stride=1, pad=0,
-                 activation=relu.relu, wscale=1, conv_init=None,
-                 bias_init=None):
+                 activation=relu.relu, *args, **kwargs):
+
+        # If `args` is not empty, users assume the API for v1 and
+        # specifie `wscale` as a positonal argument, which we want
```
Comment (unnonouno) [inline chainer/links/connection/mlp_convolution_2d.py]: specify

File: chainer/links/connection/mlp_convolution_2d.py [hunk 3]
```diff
@@ -33,26 +36,44 @@ class MLPConvolution2D(link.ChainList):
...
+        # to detect and forbid with an explicit error message.
+        msg = ('wscale is not supported anymore. '
+               'Use conv_init and bias_init argument to change '
+               'the scale of initial parameters.')
+        if args:
```
Comment (unnonouno) [inline chainer/links/connection/mlp_convolution_2d.py]: OK, this is safer than my implementation. Do you have a plan to make a method to support this type of warning?

Comment (delta2323) [inline chainer/links/connection/mlp_convolution_2d.py]: I am thinking of it. But I will do that after we need to do similar things in other functions.

Comment (unnonouno) [inline chainer/links/connection/mlp_convolution_2d.py]: OK

**Comment by unnonouno** [comment]
LGTM

**Pull request merged**